ocaml/typing/typecore.ml

```diff
@@ -3781,18 +3785,20 @@ and type_approx_aux_jane_syntax : Jane_syntax.Expression.t -> _ = function
   | Jexp_immutable_array _
   | Jexp_unboxed_constant _ -> ()
 
-and type_approx_aux_case_rhs env rhs in_function ty_expected =
-  match rhs with
-  | Psimple_rhs e | Pboolean_guarded_rhs { pbg_rhs = e; _ } ->
-      type_approx_aux env e in_function ty_expected
-  | Ppattern_guarded_rhs { ppg_cases; _ } ->
-      type_approx_aux_cases env ppg_cases in_function ty_expected
```

Comment:
As discussed, let's introduce the `Psimple_rhs ... | ...` type into this file so we can minimize the diff.

Reply:
Done, although we can further reduce the diff by using the original field names. I opted to omit the prefixes (like `pbg_`) for now. What do you think?